internal/internal_worker_base.go

```diff
@@ -323,15 +351,14 @@ func (bw *baseWorker) runTaskDispatcher() {
 		case <-bw.stopCh:
 			return
 		case task := <-bw.taskQueueCh:
-			// for non-polled-task (local activity result as task), we don't need to rate limit
+			// for non-polled-task (local activity result as task or eager task), we don't need to rate limit
 			_, isPolledTask := task.(*polledTask)
 			if isPolledTask && bw.taskLimiter.Wait(bw.limiterContext) != nil {
 				if bw.isStop() {
 					return
 				}
 			}
-			bw.stopWG.Add(1)
-			go bw.processTask(task)
+			bw.processTaskAsync(task, func() {})
```

Comment:
```suggestion
			bw.processTaskAsync(task, nil)
```

And handle nil on the `processTaskAsync` side. Don't have to and definitely a probably-not-needed micro-optimization, but this is called for every task.